src/main/java/nl/delftelectronics/spaceinvaders/core/entities/Barricade.java

```diff
@@ -9,6 +11,8 @@
 public class Barricade extends SpriteEntity implements Collidable {
     private static final String FILENAME = "/barricade.png";
 
+    private EventListenerList destroyedEvent;
```

Comment:
That would require type checking and casting for the sake of code reuse. Implementing another listener prevents this.